Zend/zend.c

```diff
@@ -2005,6 +2020,25 @@ ZEND_API void *zend_map_ptr_new(void)
 	return ZEND_MAP_PTR_PTR2OFFSET(ptr);
 }
 
+ZEND_API void *zend_map_ptr_new_static(void)
+{
+	void **ptr;
+
+	if (zend_map_ptr_static_last >= zend_map_ptr_static_size) {
+		CG(map_ptr_last) += 4096;
+		zend_map_ptr_static_size += 4096;
+		/* Grow map_ptr table */
+		CG(map_ptr_size) = ZEND_MM_ALIGNED_SIZE_EX(CG(map_ptr_last), 4096);
+		CG(map_ptr_real_base) = perealloc(CG(map_ptr_real_base), CG(map_ptr_size) * sizeof(void*), 1);
+		memmove((char*)CG(map_ptr_real_base) + 4096 * sizeof(void *), CG(map_ptr_real_base), (CG(map_ptr_last) - 4096) * sizeof(void *));
```

Comment:
```suggestion
		memmove((void**)CG(map_ptr_real_base) + 4096, CG(map_ptr_real_base), (CG(map_ptr_last) - 4096) * sizeof(void *));
```

Also, it's unlikely to matter for this case (as static allocations are only done for startup) but when moving to a new location this essentially copies the data twice, once to the new allocation, and then the `memmove`.

Reply:
Yeah, probably better to just malloc and then memcpy.